### PR TITLE
[e2e] Make tests runnable against dev

### PR DESCRIPTION
**Pull request by tanner-ricks**
Made e2e test suite runnable against the dev deployment when running locally.

## Changes

- Added calls to cleanup logic after tests
- Added code to change the way auth works when running against dev
- Added additional environment variables that were missed before
- Fixed certificate issue causing the tests to run against dev

## How to test this PR

1. Read the notes
2. Make sure all other repos are up to date
3. Pull the branch for the sbl-frontend repo
4. Update your local .env file to reflect what is in the .env.example file
5. Comment out and override the following values in your local .env with what is listed below replacing the placeholder values with their appropriate ones:
    - SBL_PLAYWRIGHT_TEST_TARGET="<FRONTEND_URL>"
    - SBL_PLAYWRIGHT_TEST_REGTECH_TARGET="<USER_FI_URL>"
    - SBL_PLAYWRIGHT_TEST_FILING_TARGET="<FILING_URL>"
    - SBL_PLAYWRIGHT_TEST_MAIL_TARGET="<MAIL_URL>"
    - SBL_PLAYWRIGHT_TEST_CLEANUP_TARGET="<CLEANUP_URL>"
    - SBL_PLAYWRIGHT_TEST_KC_TARGET="<KEYCLOAK_URL>"
    - SBL_PLAYWRIGHT_TEST_KC_CLI_CLIENT_SECRET="<KEYCLOAK_SECRET>"

## Notes

- As long as dev is out of date and the tests have been updated then it is to be expected that you will see failures when running against dev.
- As it stands at the time of this writing, there are test failures related to dev being out of date.

### CODE REVIEW

**Review by billhimmelsbach — COMMENTED** [review]
This is really great! I had a few comments. Has the migration affected the certs at all?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,19 +7,36 @@ on:
...
-  SBL_ENABLE_PLAYWRIGHT_TEST_SETTINGS: "false"
-  SBL_PLAYWRIGHT_TEST_TARGET: "http://localhost:8899"
+
+  ###############################################################
+  # ONLY ADD ENVIRONMENT VARIABLES THAT ARE NOT MEANT TO BE 
+  # PACKAGED WITH AND VISIBLE TO THE CLIENT BELOW
+  ###############################################################
+  NODE_EXTRA_CA_CERTS: "./e2e/certs/entrust_chain.crt.pem"
+  
+  SBL_ENABLE_PLAYWRIGHT_TEST_SETTINGS: false
```
[the remark this answers is not in the document]
Comment (tanner-ricks) [reply .github/workflows/test.yml]: Moved it to the public section.

File: ENV-GUIDE.md [hunk 2]
```diff
@@ -30,19 +42,40 @@ SBL_PLAYWRIGHT_TEST_KC_ADMIN_CLIENT_ID="regtech-client"
...
 - ask a devops/backend engineer to help you update the `values.yaml` overrides
 
-### To use an environment variable
+### To use a public environment variable
 
 ```js
 import.meta.env.SOME_KEY;
 ```
+
+### To add a new private environment variable
```
[the remark this answers is not in the document]
Comment (tanner-ricks) [reply ENV-GUIDE.md:68]: Added new verbiage.

**Comment by billhimmelsbach** [comment]
Jason is going to work on a fix for the cleanup-api, then this should be good to go.

**Review by billhimmelsbach — APPROVED** [review]
Approved! Thanks for fixing that issue with the Complete user profile page test! Mind making a ticket for reenabling the error?

**Pull request merged**